core/database.py

```
@@ -11,12 +11,22 @@
+        logger.info("Database connection pool already initialized; skipping re-initialization.")
+        return
+
+    created_pool = None
     try:
-        pool = await asyncpg.create_pool(POSTGRES_URL)
+        created_pool = await asyncpg.create_pool(POSTGRES_URL)
+        pool = created_pool
         logger.info("Database connection pool created.")
         await create_schema()
     except Exception as e:
         logger.error(f"Failed to initialize database: {e}")
```

Ruff (0.15.2)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)